### PR TITLE
Parse supported verbs, only process types that support list/watch for informer/listergen

### LINKED ISSUE #40: informers: don't generate if the type is not list+watchable
e.g. TokenReviewInformer, SubjectAccessReviewInformer, etc.

### PR DESCRIPTION
**Pull request by fabianvf**
closes #40

### CODE REVIEW

File: pkg/generators/listergen/listergen.go [hunk 1]
```diff
@@ -104,112 +118,87 @@ func (g Generator) Run(ctx *genall.GenerationContext, f flag.Flags) error {
...
-		g.inputDir = f.InputDir
-		pkg, hasGoMod := util.CurrentPackage(f.InputDir)
-		if len(pkg) == 0 {
-			return fmt.Errorf("error finding the module path for this package %q", f.InputDir)
-		}
-		g.inputpkgPaths = pkgPaths{
-			basePackage: pkg,
-			hasGoMod:    hasGoMod,
-		}
+	if err := flag.ValidateFlags(f); err != nil {
```
Comment (ncdc) [inline pkg/generators/listergen/listergen.go:121]: Haven't you already validated flags in Run before calling setDefaults?

File: pkg/parser/extract.go [hunk 2]
```diff
@@ -83,7 +85,7 @@ func GetGV(f flag.Flags) ([]types.GroupVersions, error) {
 	return groupVersions, nil
 }
 
-func GetGVKs(ctx *genall.GenerationContext, inputDir string, groupVersions []types.GroupVersions) (map[Group]map[types.PackageVersion][]Kind, error) {
+func GetGVKs(ctx *genall.GenerationContext, inputDir string, groupVersions []types.GroupVersions, requiredVerbs []string) (map[Group]map[types.PackageVersion][]Kind, error) {
```
Comment (ncdc) [inline pkg/parser/extract.go:88]: (no action required) if you make requiredVerbs a sets.String, you can invoke HasAll on it

Comment (fabianvf) [reply pkg/parser/extract.go:88]: where is `sets` from? my editor grabs `"sigs.k8s.io/kustomize/kyaml/sets"`, which I assume isn't the right one

**Comment by fabianvf** [comment]
addressed (other than the sets one)

File: pkg/parser/extract.go (hunk 2 above)
Comment (fabianvf) [reply pkg/parser/extract.go:88]: hmm gave it a go but it ended up being a little less clear to me than just using the map[string]bool and []string

File: pkg/generators/listergen/listergen.go [hunk 3]
```diff
@@ -104,112 +119,83 @@ func (g Generator) Run(ctx *genall.GenerationContext, f flag.Flags) error {
...
-		pkg, hasGoMod := util.CurrentPackage(f.InputDir)
-		if len(pkg) == 0 {
-			return fmt.Errorf("error finding the module path for this package %q", f.InputDir)
-		}
-		g.inputpkgPaths = pkgPaths{
-			basePackage: pkg,
-			hasGoMod:    hasGoMod,
-		}
+	g.inputDir = f.InputDir
+	absoluteInputDir, err := filepath.Abs(g.inputDir)
```
Comment (varshaprasad96) [inline pkg/generators/listergen/listergen.go:123]: Not sure if its already been tested, but there was an issue with using absolute path when APIs were defined in a sub-module and the generator was run from a parent directory. That was the reason it was reverted [here](https://github.com/kcp-dev/code-generator/pull/30/files). But please feel free to ignore it if it behaves as required.

Comment (fabianvf) [reply pkg/generators/listergen/listergen.go]: must have copied it to informergen before the reversion, changed it for lister and informergen

**Review by ncdc — APPROVED** [review]

**Pull request merged**